### ISSUE TITLE
Code splitting doesn't work with dependencies

### ISSUE THREAD

**Issue by Nick-Mazuk** [sveltejs/kit#1831]
**Describe the bug**

3rd-party dependencies don't follow code-splitting. If you use a 3rd-party dependency on any page, it's loaded on every page.

This behavior is egregious for necessary, but large dependencies that are used on only a few pages.

**To Reproduce**

Reproduction repo: https://github.com/Nick-Mazuk/sveltekit-bundle-size

- Create new Svelte/kit project
- Add a new component that imports any dependency from npm (e.g., Lodash)
- Create a new page and import that component
- Build the project (with pretty much any adpater)
- Go to the main page (e.g., the one that doesn't load the component)
- In Chrome Dev tools, look at the code in the sources tab `Sources > domain > _app > chunks > vendor-12352414.js`
  - exact vendor.js file will differ because the hash may change
  - In this file, you can see the entire source for the npm dependency (e.g., Lodash)

This behavior seems to occur with every npm package and every Sveltekit adapter.

**Expected behavior**

3rd-party dependencies should not be loaded on every page—only the ones they're needed for. Otherwise, Sveltekit isn't doing the "code splitting" as advertised.

At the very least, large dependencies should be code split. For instance, three.js (600kb) should not be bundled into the main vendor bundle. It should only be loaded on the necessary pages. Since projects likely have hundreds of small 1-10kb dependencies, code splitting each of these could lead to many useless HTTP requests. But by splitting the large dependencies out of the main vendor bundle, you can at least get the performance benefits of not loading those large dependencies on every page.

**Information about your SvelteKit Installation:**

<details>
  <summary>Diagnostics</summary>
<pre><code>  System:
    OS: macOS 11.4
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 12.35 GB / 64.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.2.0 - ~/.nvm/versions/node/v16.2.0/bin/node
    Yarn: 1.22.10 - ~/.nvm/versions/node/v16.2.0/bin/yarn
    npm: 7.13.0 - ~/.nvm/versions/node/v16.2.0/bin/npm
  Browsers:
    Brave Browser: 91.1.26.74
    Firefox: 88.0.1
    Safari: 14.1.1
  npmPackages:
    @sveltejs/adapter-node: ^1.0.0-next.30 => 1.0.0-next.30 
    @sveltejs/adapter-static: ^1.0.0-next.13 => 1.0.0-next.13 
    @sveltejs/adapter-vercel: ^1.0.0-next.23 => 1.0.0-next.23 
    @sveltejs/kit: next => 1.0.0-next.120 
    svelte: ^3.34.0 => 3.38.3</code></pre>
</details>

Tested with multiple adapters:

- No adapter
- Node
- Static
- Cloudflare
- Vercel

**Severity**

It prohibits using any larger dependencies, or even many small dependencies on multiple pages. Because if you use something like three.js (600kb) on one page for awesome animations, it's now loaded on every single page.

Another popular repo is [zxvbn](https://github.com/dropbox/zxcvbn) (300kb) to ensure users are creating strong passwords. This should only load on the "create account" page, but now it's loaded on every… single… page.

In short, really annoying.

**Additional context**

I know this example does use the "bad practice" of importing all of Lodash (use ESM for tree-shaking). What's more important is that it easily demonstrates this behavior.

**Comment by istarkov** [comment]
Can you reproduce this without prefetch? https://github.com/Nick-Mazuk/sveltekit-bundle-size/blob/main/src/routes/index.svelte#L4

**Comment by Nick-Mazuk** [comment]
> Can you reproduce this without prefetch? https://github.com/Nick-Mazuk/sveltekit-bundle-size/blob/main/src/routes/index.svelte#L4

Yes. Just pushed a new version of the repo without the prefetch and I'm seeing the same things,

**Comment by philip-weber** [comment]
Is this a duplicate to #1632?
